pyogrio/tests/test_geopandas_io.py

```diff
@@ -132,13 +135,52 @@ def test_read_layer_invalid(naturalearth_lowres_all_ext):
 @pytest.mark.filterwarnings("ignore: Measured")
 def test_read_datetime(test_fgdb_vsi):
     df = read_dataframe(test_fgdb_vsi, layer="test_lines", max_features=1)
-    if Version(pd.__version__) >= Version("2.0.0"):
+    if PANDAS_GE_20:
         # starting with pandas 2.0, it preserves the passed datetime resolution
         assert df.SURVEY_DAT.dtype.name == "datetime64[ms]"
     else:
         assert df.SURVEY_DAT.dtype.name == "datetime64[ns]"
 
 
+def test_read_datetime_tz(test_datetime_tz, tmp_path):
+    df = read_dataframe(test_datetime_tz)
+    raw_expected = ["2020-01-01T09:00:00.123-05:00", "2020-01-01T10:00:00-05:00"]
+
+    if PANDAS_GE_20:
+        expected = pd.to_datetime(raw_expected, format="ISO8601").as_unit("ms")
+    else:
+        expected = pd.to_datetime(raw_expected)
+    expected = pd.Series(expected, name="col")
+
+    assert_series_equal(df.col, expected)
+    # test write and read round trips
+    fpath = tmp_path / "test.gpkg"
+    write_dataframe(df, fpath)
+    df_read = read_dataframe(fpath)
+    assert_series_equal(df_read.col, expected)
+
+
+def test_write_datetime_mixed_offset(tmp_path):
+    # Summer Time (GMT+11), standard time (GMT+10)
+    dates = ["2023-01-01 11:00:01.111", "2023-06-01 10:00:01.111"]
+    tz = pytz.timezone("Australia/Sydney")
+    ser_naive = pd.Series(pd.to_datetime(dates), name="dates")
+    ser_localised = ser_naive.dt.tz_localize(tz)
+    ser_utc = ser_localised.dt.tz_convert(pytz.UTC)
+    if PANDAS_GE_20:
+        ser_utc = ser_utc.dt.as_unit("ms")
+
+    df = gp.GeoDataFrame(
+        {"dates": ser_localised, "geometry": [Point(1, 1), Point(1, 1)]}
+    )
+    fpath = tmp_path / "test.gpkg"
+    write_dataframe(df, fpath)
+    df_local = read_dataframe(fpath)
+    # GDAL tz only encodes offsets, not timezones, for multiple offsets
+    # read as utc datetime as otherwise would be read as string
+    assert_series_equal(ser_utc, df_local["dates"])
```

Comment:
```suggestion
    result = read_dataframe(fpath)
    # GDAL tz only encodes offsets, not timezones, for multiple offsets
    # read as utc datetime as otherwise would be read as string
    assert_series_equal(result["date"], ser_utc)
```

I find `df_local` confusing as name (it's not local time?)

Reply:
Yep agree, I think this is leftover from an earlier version of the test